### PR TITLE
Fix passing of bf16 and string types to convert model().

### PR DESCRIPTION
**Pull request by popovaan**
### Details:
 - Removed internal conversion of all typed passed to `convert_model()` to `np.dtype`, as some types do not have `numpy` analogs, which causes errors. Also this step is redundant as we can use `ov.Type` instead.

### Tickets:
 - 138755

### CODE REVIEW

**Review by eaidova — APPROVED** [review]

File: tools/ovc/openvino/tools/ovc/moc_frontend/type_utils.py [hunk 1]
```diff
@@ -33,7 +33,22 @@ def to_ov_type(val):
...
+                tf.int8: ov.Type.i8,
+                tf.int16: ov.Type.i16,
+                tf.int32: ov.Type.i32,
+                tf.int64: ov.Type.i64,
+                tf.bool: ov.Type.boolean,
+                tf.string: ov.Type.string
+            }
+            if val not in tf_to_ov_type:
+                raise Exception("The provided data time is not supported {}.".format(val))
+            return tf_to_ov_type[val]
```
Comment (rkazants) [inline tools/ovc/openvino/tools/ovc/moc_frontend/type_utils.py:51]: please do not introduce new dict of types. We should already created the same one in utils. Please check

**Review by rkazants — CHANGES_REQUESTED** [review]
please re-use table of types

File: tools/ovc/openvino/tools/ovc/moc_frontend/type_utils.py (hunk 1 above)
Comment (rkazants) [reply tools/ovc/openvino/tools/ovc/moc_frontend/type_utils.py:51]: I am mistaked about existence. Let us create this dictionary in Python TF FE sources in utils.

**Review by rkazants — APPROVED** [review]

File: tools/ovc/openvino/tools/ovc/moc_frontend/type_utils.py (hunk 1 above)
Comment (popovaan) [reply tools/ovc/openvino/tools/ovc/moc_frontend/type_utils.py]: Done.

File: tools/ovc/openvino/tools/ovc/convert_impl.py [hunk 2]
```diff
@@ -361,10 +360,8 @@ def normalize_inputs(argv: argparse.Namespace):
                 shape_list.append(PartialShape(inp.shape))
             if inp.type is not None:
                 # Convert type to numpy type for uniformity of stored values
-                if isinstance(inp.type, str):
-                    data_type_list.append(destination_type_to_np_data_type(inp.type))
-                elif isinstance(inp.type, Type):
-                    data_type_list.append(inp.type.to_dtype().type)
+                if isinstance(inp.type, (np.dtype, str)):
+                    data_type_list.append(Type(inp.type))
```
[the remark this answers is not in the document]
Comment (popovaan) [reply tools/ovc/openvino/tools/ovc/convert_impl.py]: We have `to_ov_type()` method from tools/ovc/openvino/tools/ovc/moc_frontend/type_utils.py, it covers Torch, TensorFlow, PDPD. Extended it for NumPy and string case and reused it here.

**Pull request merged**